serde_arrow/src/internal/serialization/outer_sequence_builder.rs

```diff
@@ -226,7 +227,52 @@ fn build_builder(path: String, field: &Field) -> Result<ArrayBuilder> {
                 .ctx(&ctx)?,
             )
         }
-        T::Struct(children) => A::Struct(build_struct(path, children, field.nullable)?),
+        T::Struct(children) => {
+            if let Some(Strategy::EnumsWithNamedFieldsAsStructs) =
+                get_strategy_from_metadata(&field.metadata)?
+            {
+                let mut related_fields: BTreeMap<&str, Vec<Field>> = BTreeMap::new();
+                let mut builders: Vec<(ArrayBuilder, FieldMeta)> = Vec::new();
+
+                for field in children {
+                    let Some(variant_name) = field.union_variant_name() else {
+                        // TODO: failure message
```

Comment:
probably better to use [`todo!()`](https://doc.rust-lang.org/stable/std/macro.todo.html) in these cases. This way you can easily search for it and still have the correct runtime behavior.

Reply:
fixed